src/main/java/edu/kit/datamanager/metastore2/health/MetadataRepoHealthCheck.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2023 Karlsruhe Institute of Technology.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.kit.datamanager.metastore2.health;
+
+import edu.kit.datamanager.metastore2.configuration.MetastoreConfiguration;
+import edu.kit.datamanager.metastore2.dao.ILinkedMetadataRecordDao;
+import java.io.File;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+import static org.hibernate.tool.schema.SchemaToolingLogging.LOGGER;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.boot.actuate.info.Info;
+import org.springframework.stereotype.Component;
+
+/**
+ *
+ * @author hartmann-v
+ */
+@Component("MetadataRepo")
+public class MetadataRepoHealthCheck extends HealthCheck  {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MetadataRepoHealthCheck.class);
+  
+  private final MetastoreConfiguration metadataConfig;
+
+  @Autowired
+  private final ILinkedMetadataRecordDao metadataRecordDao;
+
+  /**
+   *
+   * @param metadataConfig
+   * @param metadataRecordDao
+   */
+  public MetadataRepoHealthCheck(MetastoreConfiguration metadataConfig,
+          ILinkedMetadataRecordDao metadataRecordDao) {
+    this.metadataConfig = metadataConfig;
+    this.metadataRecordDao = metadataRecordDao;
+  }
+
+  @Override
+  public Health health() {
+    LOG.trace("Check for MetadataRepo health information...");
+
+    URL basePath = metadataConfig.getBasepath();
+    Map<String, String> details = testDirectory(basePath);
+
+    if (details.isEmpty()) {
+      return Health.down().withDetail("No of metadata documents", 0).build();
+    } else {
+      details.put("No of metadata documents", Long.toString(metadataRecordDao.count()));
+      return Health.up().withDetails(details).build();
+    }
+  }
+
+  @Override
+  public void contribute(Info.Builder builder) {
```

Comment:
Why putting the same information into the InfoActuator?

Reply:
This is meant to be a basis for discussion.
I think it's good to see this information in the /actuator/info section. 
Not absolutely necessary for /actuator/health, but if you already have it...